modules/upload/upload.go

```diff
@@ -75,18 +75,28 @@ func AddUploadContext(ctx *context.Context, uploadType string) {
 	if uploadType == "release" {
 		ctx.Data["UploadUrl"] = ctx.Repo.RepoLink + "/releases/attachments"
 		ctx.Data["UploadRemoveUrl"] = ctx.Repo.RepoLink + "/releases/attachments/remove"
+		ctx.Data["UploadLinkUrl"] = ctx.Repo.RepoLink + "/releases/attachments"
 		ctx.Data["UploadAccepts"] = strings.Replace(setting.Repository.Release.AllowedTypes, "|", ",", -1)
 		ctx.Data["UploadMaxFiles"] = setting.Attachment.MaxFiles
 		ctx.Data["UploadMaxSize"] = setting.Attachment.MaxSize
 	} else if uploadType == "comment" {
 		ctx.Data["UploadUrl"] = ctx.Repo.RepoLink + "/issues/attachments"
 		ctx.Data["UploadRemoveUrl"] = ctx.Repo.RepoLink + "/issues/attachments/remove"
+		ctx.Data["UploadLinkUrl"] = ctx.Repo.RepoLink + "/issues/attachments"
+		ctx.Data["UploadAccepts"] = strings.Replace(setting.Attachment.AllowedTypes, "|", ",", -1)
+		ctx.Data["UploadMaxFiles"] = setting.Attachment.MaxFiles
+		ctx.Data["UploadMaxSize"] = setting.Attachment.MaxSize
+	} else if uploadType == "editcomment" {
```

Comment:
does it really need a new uploadType? Why not add a optional boolean argument `isEdit` or something?

Reply:
yes

* ctx.Params(":index") will not be present otherwise. 
* isEdit makes no sense for any of the other `"uploadtype"` they don't have edits.
* `"uploadtype"` is just an argument to this function - it's not set anywhere.

Reply:
When or if we require isEdit to have a different effect for release and repo then we can refactor this.

In the meantime it's genuinely a different type of upload.

Reply:
Releases can certainly edit, but I think it's been broken or never been implemented (https://github.com/go-gitea/gitea/issues/12963).

Reply:
How about still adding that boolean and a `if` around setting of `UploadLinkUrl`? I prefer to avoid duplicating the whole clause.

Reply:
I don't see how this is an improvement but fine.

Reply:
It prevents unneccary overloading of the `uploadType` variable with two characteristics (it's type and if it's an create/edit). I think it's easier to extend later on if we have one variable per characteristic.